server/src/routes/user.ts

```diff
@@ -0,0 +1,57 @@
+import { prisma } from "../common";
+import express from "express";
+import { asyncHandler } from "../utils/asyncHandler";
+
+export const userRoutes = express.Router();
+
+// Filter using query string in url with parameters role and category
+userRoutes.route("/").get(
+  asyncHandler(async (req, res) => {
+    let categoryFilter = (req.query.category as string) || null;
+    let roleFilter = (req.query.role as string) || null;
+    let filter: any = {};
+
+    if (categoryFilter !== null) {
+      const categoryId = parseInt(categoryFilter);
+      filter.categoryGroupId = categoryId;
+    }
+
+    if (roleFilter !== null) {
+      roleFilter = roleFilter.toUpperCase();
+      filter.role = roleFilter;
+    }
+
+    const users = await prisma.user.findMany({
+      where: filter,
+    });
+
+    let usersResponse: any = [];
+    users.forEach(user => {
+      const { token, ...userWithoutToken } = user;
+      usersResponse.push(userWithoutToken);
+    });
+    res.status(200).json(usersResponse);
+  })
+);
+
+// Update user
+userRoutes.route("/:id").patch(
+  asyncHandler(async (req, res) => {
+    let data: any = {};
+    Object.keys(req.body).forEach(key => {
+      if (["role", "categoryGroup"].includes(key)) {
+        data[key] = req.body[key];
+      }
+    });
+
+    const updatedUser = await prisma.user.update({
+      where: {
+        uuid: req.params.id,
+      },
+      data: data,
+    });
+
+    const { token, ...userWithoutToken } = updatedUser;
```

Comment:
Since you are doing this same function of removing the token here and in the other handler, I'd extract this functionality into a utility function like `sanitize` or something that will remove the user field. That way in the future, if we have to add a new secret column, or add a new route, we don't have to keep repeating this functionality.